### PR TITLE
Practice5

### PR DESCRIPTION
**Pull request by nick-krasnitskiy**
1. Create UI for the main screen (map view and search bar).
2. Create user gesture tap recognize and search by the city name. 
4. Create a map moving to the searched element.
5. Create UI for the second screen (detail weather screen) (open by a tap on the pin)
6. Add a custom pin image. Implement a callout display after the image loading. Add an info button for open detail VC.
7. Implement the current weather display on the second screen.
8. Add weekly forecast for the second screen.
9. Add settings (temperature units) stored into the NSUserDefaults and update map view annotation.

### CODE REVIEW

File: DogsList/.swiftlint.yml [hunk 1]
```diff
@@ -3,10 +3,6 @@ disabled_rules: # rule identifiers turned on by default to exclude from running
   - colon
```
Comment (vasilevkin) [inline DogsList/.swiftlint.yml]: do not disable swiftlint rules
it's not a good practice
enable all rules and update your code to comply the linter rules

File: DogsList/DogsList.xcodeproj/xcuserdata/nvkrasnitskiy.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 2]
```diff
@@ -7,7 +7,7 @@
 		<key>DogsList.xcscheme_^#shared#^_</key>
```
Comment (vasilevkin) [inline DogsList/DogsList.xcodeproj/xcuserdata/nvkrasnitskiy.xcuserdatad/xcschemes/xcschememanagement.plist:7]: this file should not be in review

File: DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+enum NewSection: Int, CaseIterable {
+    case first
+    case second
+    case third
+    
+    var groupHeight: NSCollectionLayoutDimension {
+        switch self {
+        case .first:
+            return .fractionalHeight(0.2)
```
Comment (vasilevkin) [inline DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift]: avoid magic numbers

File: DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    
+}
+
+class AnimalListViewController: UIViewController {
+    
+    @IBOutlet private weak var collectionView: UICollectionView!
+    private let indicator = UIActivityIndicatorView(style: .large)
+  
+    typealias DataSource = UICollectionViewDiffableDataSource<NewSection, Animal>
+    private var dataSource: DataSource!
```
Comment (vasilevkin) [inline DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift]: avoid force unwrap optionals

File: DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+        let layout = UICollectionViewCompositionalLayout { (sectionIndex, _) -> NSCollectionLayoutSection? in
+            
+            guard let sectionType = NewSection(rawValue: sectionIndex) else { fatalError("unknown section kind") }
+        
+            let heights = sectionType.groupHeight
+            let widths = sectionType.groupWidth
+            
+            let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1.0), heightDimension: .fractionalHeight(1.0))
+            let item = NSCollectionLayoutItem(layoutSize: itemSize)
+            item.contentInsets = NSDirectionalEdgeInsets(top: 2, leading: 2, bottom: 2, trailing: 2)
```
Comment (vasilevkin) [inline DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift]: avoid magic numbers

**Review by vasilevkin — CHANGES_REQUESTED** [review]
please check the pr
it should contain your changes for Practice 5 only
please exclude all other files/changes not related to Practice 5
and fix all SwiftLint warnings and errors

File: DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift (hunk 3 above)
Comment (nick-krasnitskiy) [reply DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift]: Fixed.

File: DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift (hunk 4 above)
Comment (nick-krasnitskiy) [reply DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift]: Fixed.

File: DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift (hunk 5 above)
Comment (nick-krasnitskiy) [reply DogsList/DogsList/Modules/AnimalList/Controllers/AnimalListViewController.swift]: Fixed.

File: DogsList/.swiftlint.yml (hunk 1 above)
Comment (nick-krasnitskiy) [reply DogsList/.swiftlint.yml]: Fixed.

**Review by vasilevkin — APPROVED** [review]
LGTM

**Pull request merged**